network/transport/protocol.go

```diff
@@ -36,4 +36,20 @@ type Protocol interface {
 	PeerDiagnostics() map[PeerID]Diagnostics
 	// Version returns the version of the protocol
 	Version() int
+	// MethodName returns the fully qualified name of the gRPC stream.
+	MethodName() string
+}
+
+type Connection interface {
+	// Send tries to send the given message over the stream of the given protocol.
```

Comment:
I'm not sure this belongs in the transport.Connection interface (move to grpc.Connection iface?)

Reply:
moved back to `grpc.Connection` and unexported `closeError` in that interface